src/core.c/CompUnit/RepositoryRegistry.pm6

```diff
@@ -285,10 +285,6 @@ class CompUnit::RepositoryRegistry {
     }
 
     method run-script($script, :$name, :$auth, :$ver, :$api) {
-        shift @*ARGS if $name;
-        shift @*ARGS if $auth;
-        shift @*ARGS if $ver;
-
```

Comment:
This change probably isn't sufficient -- it would need to remove `raku-dist-name`, `raku-dist-ver`, etc from `@*ARGS` so they don't get used by the script it will run.

Of course the current logic is terribly broken as well since it incorrectly assumes the order of `@*ARGS` and will happily remove parameters such as `foo` or `bar` instead of `name` and `auth` based on the order.